src/server/bitops_family.cc

```diff
@@ -0,0 +1,302 @@
+      return OpStatus::WRONG_TYPE;
+    }
+    bool reset = false;
+    std::string existing_entry{GetString(shard_, it->second)};
+    if ((existing_entry.size() * OFFSET_FACTOR) <= offset) {  // need to resize first
+      existing_entry.resize(GetByteIndex(offset) + 1, 0);
+      reset = true;
+    }
+    old_value = SetBitValue(offset, bit_value, &existing_entry);
+    if (reset || old_value != bit_value) {  // we made a "real" change to the entry, save it
+      UpdateBitMapValue(existing_entry);
+    }
+  }
+  return old_value;
+}
+
+// ------------------------------------------------------------------------- //
+//  Impl for the command functions
+void BitPos(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitField(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitOp(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendOk();
+}
+
+void GetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "GETBIT key offset"
+  // see https://redis.io/commands/getbit/
+
+  uint32_t offset{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return ReadValueBitsetAt(t->GetOpArgs(shard), key, offset);
+  };
+  Transaction* trans = cntx->transaction;
+  OpResult<bool> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    DVLOG(2) << "GET" << trans->DebugId() << "': key: '" << key << ", value '" << result.value()
+             << "'\n";
+    // we have the value, now we need to get the bit at the location
+    long val = result.value() ? 1 : 0;
+    (*cntx)->SendLong(val);
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      default:
+        DVLOG(2) << "GET " << key << " nil";
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+    }
+  }
+}
+
+void SetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "SETBIT key offset new_value"
+  // see https://redis.io/commands/setbit/
+
+  uint32_t offset{0};
+  int32_t value{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset) || !absl::SimpleAtoi(ArgS(args, 3), &value)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    OverrideValue set_operation{t->GetOpArgs(shard)};
+
+    return set_operation.Set(key, offset, value != 0);
+  };
+
+  Transaction* trans = cntx->transaction;
+  OpResult<bool> result = trans->ScheduleSingleHopT(std::move(cb));
+  if (result) {
+    long res = result.value() ? 1 : 0;
```

Comment:
in c++ you do not need to cast bool to long, `(*cntx)->SendLong(*result);` works

Reply:
OK, again maybe 2 1/2 years with a language that force you to do it, made me be over reacting to this - I will make sure in the future to only do it when the compiler don't like it